misk/src/test/kotlin/misk/web/interceptors/RequestLoggingInterceptorTest.kt

```diff
@@ -0,0 +1,174 @@
+import misk.logging.LogCollectorModule
+import misk.scope.ActionScoped
+import misk.security.authz.AccessControlModule
+import misk.security.authz.FakeCallerAuthenticator
+import misk.security.authz.MiskCallerAuthenticator
+import misk.security.authz.Unauthenticated
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.Get
+import misk.web.PathParam
+import misk.web.ResponseContentType
+import misk.web.WebTestingModule
+import misk.web.actions.WebAction
+import misk.web.actions.WebActionEntry
+import misk.web.actions.asChain
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import okhttp3.OkHttpClient
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+import kotlin.random.Random
+
+@MiskTest(startService = true)
+internal class RequestLoggingInterceptorTest {
+
+  @MiskTestModule
+  val module = TestModule()
+  val httpClient = OkHttpClient()
+
+  @Inject private lateinit var jettyService: JettyService
+  @Inject private lateinit var logCollector: LogCollector
+  @Inject private lateinit var includeBodyRequestLoggingAction: IncludeBodyRequestLoggingAction
+
+  @Test
+  fun includesBody() {
+    assertThat(invoke("/call/includeBodyRequestLogging/hello", "caller").isSuccessful)
+    val messages = logCollector.takeMessages(RequestLoggingInterceptor::class)
+    assertThat(messages).hasSize(2)
+    assertThat(messages.get(0)).isEqualTo(
+      "IncludeBodyRequestLoggingAction principal=caller request=[hello]")
+    assertThat(messages.get(1)).matches(
+      "IncludeBodyRequestLoggingAction principal=caller time=.+? response=echo: hello"
+    )
+  }
+
+  @Test
+  fun excludesBody() {
+    assertThat(invoke("/call/excludeBodyRequestLogging/hello", "caller").isSuccessful)
+    val messages = logCollector.takeMessages(RequestLoggingInterceptor::class)
+    assertThat(messages).hasSize(2)
+    assertThat(messages.get(0)).isEqualTo(
+      "ExcludeBodyRequestLoggingAction principal=caller request=")
+    assertThat(messages.get(1)).matches(
+      "ExcludeBodyRequestLoggingAction principal=caller time=.+? response="
+    )
+  }
+
+  @Test
+  fun exceptionThrown() {
+    assertThat(invoke("/call/exceptionThrowingRequestLogging/fail", "caller").code()).isEqualTo(500)
+    val messages = logCollector.takeMessages(RequestLoggingInterceptor::class)
+    assertThat(messages).hasSize(2)
+    assertThat(messages.get(0)).isEqualTo(
+      "ExceptionThrowingRequestLoggingAction principal=caller request=[fail]")
+    assertThat(messages.get(1)).matches(
+      "ExceptionThrowingRequestLoggingAction principal=caller time=.+? failed"
+    )
+  }
+
+  @Test
+  fun notSampled() {
+    val interceptor = RequestLoggingInterceptor(
+      action = IncludeBodyRequestLoggingAction::call.asAction(),
+      sampling = 0.4,
+      includeBody = true,
+      caller = ActionScoped.of(MiskCaller("caller")),
+      rng = FakeRandom(0.5)
```

Comment:
nb. don't yet have an injected random number provider

Reply:
seems easy enough to just build now

Reply:
done. it's a bit more nuanced with ThreadLocalRandom but got some feedback on the approach I've taken